workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+
+`setup()` is run only once, at the beginning. `draw()` on the other hand, is run repeatedly after `setup()` finishes, and in this way, provides the basis for any animation or interaction you see in your project.
+
+Let's save and refresh Live Preview.
+
+It looks like nothing, because our functions do nothing. Let's add something for `setup()` to set up.
+
+```js
+function setup() {
+  createCanvas(480,600);
+}
+```
+
+[`createCanvas()`](http://p5js.org/reference/#p5/createCanvas) is a function that takes two arguments (two numbers for width and height, respectively) and creates an [HTML canvas element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas) (i.e., the thing that will house your project) of that size.
+
+In this project, we'll be creating a tiling of overlapping circles.
+
+First, let's decide how many circles we want in each row. We can store this number in a variable **at the top, above both functions**:
+
+```js
+var NUM_CIRCLES = 12;
+
+function setup() {
+  createCanvas(480,600);
+}
+
+function draw() {
+}
+```
+
+We'll use this to determine the diameter of the circles we want to draw, by doing some simple math. Let's add the declaration at the top of the file, underneath `NUM_CIRCLES`, and define our variable within `setup()`:
+
+```js
+var NUM_CIRCLES = 12;
+var CIRCLE_DIA;
+
+function setup() {
+  createCanvas(480,600);
+  CIRCLE_DIA = width/NUM_CIRCLES;
+}
+```
+
+p5.js stores the width of the canvas in a constant named `width`. By dividing the width by the number of circles, we can calculate the length of the diameter, which we'll store in `CIRCLE_DIA`.
+
+## Part III: Drawing on the Canvas
+
+### Drawing One Circle
+
+p5.js makes drawing ellipses and circles easy with the function [`ellipse()`](http://p5js.org/reference/#p5/ellipse).
+
+Let's see the `ellipse()` function in action by drawing a circle in the middle of the screen:
+
+```js
+function draw() {
+  ellipse(width/2,height/2,CIRCLE_DIA,CIRCLE_DIA);
+}
+```
+
+Save and refresh Live Preview.
+
+In this example, the first two arguments we pass to the ellipse function are the x and y coordinates of the center of the ellipse. We've passed in `width/2` and `height/2`, respectively, which means the center of the ellipse will also be the center of the canvas. The latter two arguments are the x-width and y-width of the ellipse. Since we want to draw a circle, we're passing the same value for both.
+
+You can play around with these values to get a better feel of how this function works.
+
+NOTE TO SELF: is this necessary. should we just be focusing on the pattern instead of exploring p5.js
+
+### Drawing a Row of Circles
+
+Since we want to draw circles repeatedly, we have to write `ellipse()` as many times as we want circles.
+
+Just kidding! There's a construct in programming called a for-loop, and it repeats a set of instructions as many times as you decide.
+
+Let's remove our previous line and add a for-loop into our `draw()` function, like so:
```

Comment:
This is an amazing article – if you haven't already, @jkwok91 you need to
give this a read. Thanks so much @shiffman (and @tchoi8 for creating it).

On Fri, May 6, 2016 at 4:21 PM, Daniel Shiffman <notifications@github.com>
wrote:

> In workshops/geometric_pattern/README.md
> <https://github.com/hackclub/hackclub/pull/823#discussion_r62403675>:
>
> > +
> > +Save and refresh Live Preview.
> > +
> > +In this example, the first two arguments we pass to the ellipse function are the x and y coordinates of the center of the ellipse. We've passed in `width/2` and `height/2`, respectively, which means the center of the ellipse will also be the center of the canvas. The latter two arguments are the x-width and y-width of the ellipse. Since we want to draw a circle, we're passing the same value for both.
> > +
> > +You can play around with these values to get a better feel of how this function works.
> > +
> > +NOTE TO SELF: is this necessary. should we just be focusing on the pattern instead of exploring p5.js
> > +
> > +### Drawing a Row of Circles
> > +
> > +Since we want to draw circles repeatedly, we have to write `ellipse()` as many times as we want circles.
> > +
> > +Just kidding! There's a construct in programming called a for-loop, and it repeats a set of instructions as many times as you decide.
> > +
> > +Let's remove our previous line and add a for-loop into our `draw()` function, like so:
>
> @tchoi8 <https://github.com/tchoi8> has a great flipbook GIF for
> explaining how the draw() loop works as part of his signing coders
> project:
>
> http://taeyoonchoi.com/signing-coders-1/
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/823/files/2767b33509cc74ce1d0c1df9be40da8c8fcf5f5a#r62403675>
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>